src/psyclone/domain/lfric/kernel_interface.py

```diff
@@ -0,0 +1,412 @@
+        self._symbol_table.add(arg)
+        self._arglist.append(arg)
+
+    def mesh_ncell2d(self, var_accesses=None):
+        ''' Not implemented '''
+        raise NotImplementedError("mesh_ncell2d not implemented")
+
+    def cell_map(self, var_accesses=None):
+        ''' Not implemented '''
+        raise NotImplementedError("cell_map not implemented")
+
+    def field_vector(self, argvect, var_accesses=None):
+        ''' Implemented '''
+        mapping = {
+            "integer": IntegerVectorFieldDataDataSymbol,
+            "real": RealVectorFieldDataDataSymbol,
+            "logical": LogicalVectorFieldDataDataSymbol}
+        for idx in range(argvect.vector_size):
+            try:
+                field_data = mapping[argvect.intrinsic_type](
+                    "{0}_v{1}".format(argvect.name, idx), [1],
+                    argvect.function_space.orig_name,
+                    interface=ArgumentInterface(
+                        INTENT_MAPPING[argvect.intent]))
+            except KeyError:
+                raise NotImplementedError(
+                    "vector field of type {0} not implemented"
+                    "".format(argvect.intrinsic_type))
+
+            self._symbol_table.add(field_data)
+            self._arglist.append(field_data)
+            self._fields_list.append(field_data)
+
+    def field(self, arg, var_accesses=None):
+        ''' Create an LFRic field data object '''
+        mapping = {
+            "integer": IntegerFieldDataDataSymbol,
+            "real": RealFieldDataDataSymbol,
+            "logical": LogicalFieldDataDataSymbol}
+        try:
+            field_data = mapping[arg.intrinsic_type](
+                arg.name, [1], arg.function_space.orig_name,
+                interface=ArgumentInterface(INTENT_MAPPING[arg.intent]))
+        except KeyError:
+            raise NotImplementedError(
+                "field of type {0} not implemented"
+                "".format(arg.intrinsic_type))
+
+        self._symbol_table.add(field_data)
+        self._arglist.append(field_data)
+        self._fields_list.append(field_data)
+
+    def stencil_unknown_extent(self, arg, var_accesses=None):
+        ''' Not implemented '''
+        raise NotImplementedError("stencil_unknown_extent not implemented")
+
+    def stencil_unknown_direction(self, arg, var_accesses=None):
+        ''' Not implemented '''
+        raise NotImplementedError("stencil_unknown_direction not implemented")
+
+    def stencil(self, arg, var_accesses=None):
+        ''' Not implemented '''
+        raise NotImplementedError("stencil not implemented")
+
+    def operator(self, arg, var_accesses=None):
+        ''' Create LFRic size and operator data objects '''
+
+        op_size = OperatorSizeDataSymbol(
+            "ncell_3d", interface=self.read_access)
+        self._symbol_table.add(op_size)
+        self._arglist.append(op_size)
+        op_arg = OperatorDataSymbol(
+            arg.name, [1, 1, 1],
```

Comment:
How come `dims=[1,1,1]` here? In fact, does it make sense to specify dims to the constructor? It should be able to do that internally assuming it is passed all the things it depends on.

Reply:
Thanks to your suggestion it is no longer like this.